### PR TITLE
AutoSizeInput: Move to @grafana/ui

### LINKED ISSUE #48615: Autosize Input: Move to grafana/ui

### PR DESCRIPTION
**Pull request by ivanahuckova**
**What this PR does / why we need it**:

This PR moves AutorSizeInput from `prometheus/querybuilder/shared` components to `@grafana/ui`. I have then exported it and used it at the places, where it was previously used. I have created storybook for it. 

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana/issues/48615

### CODE REVIEW

File: packages/grafana-ui/src/components/AutoSizeInput/AutoSizeInput.mdx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+## Usage in forms with Field
+
+`AutoSizeInput` should be used with the `Field` component to get labels and descriptions. It can also be used for validation by using the `required` attribute. See the `Field` component for more information.
+
+```jsx
+<Field label="Important information" description="This information is very important, so you really need to fill it in">
+  <AutoSizeInput name="importantInput" required />
+</Field>
+```
```
Comment (achatterjee-grafana) [inline packages/grafana-ui/src/components/AutoSizeInput/AutoSizeInput.mdx:30]:
```suggestion
<Field label="Required information" description="This information is required, please enter the information.">
		 <AutoSizeInput name="importantInput" required />
		</Field>
```
```

**Review by achatterjee-grafana — COMMENTED** [review]
Added a few suggestions.

**Review by ashharrison90 — APPROVED** [review]
code looks good 🥳 

my instinct is that it should probably live in the existing `Input` subdirectory instead of having its own directory? 🤔

File: packages/grafana-ui/src/components/AutoSizeInput/AutoSizeInput.story.tsx [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+import { getAvailableIcons, IconName } from '../../types';
+import { withCenteredStory } from '../../utils/storybook/withCenteredStory';
+
+import mdx from './AutoSizeInput.mdx';
+
+const prefixSuffixOpts = {
+  None: null,
+  Text: '$',
+  ...getAvailableIcons().reduce<KeyValue<string>>((prev, c) => {
```
Comment (jackw) [inline packages/grafana-ui/src/components/AutoSizeInput/AutoSizeInput.story.tsx:15]: We have something hidden away that looks (almost) the same as required here. Maybe it's possible to reuse it? https://github.com/grafana/grafana/blob/main/packages/grafana-ui/src/utils/storybook/knobs.ts#L7-L15

**Review by jackw — APPROVED** [review]
LGTM! 🚀

File: packages/grafana-ui/src/components/AutoSizeInput/AutoSizeInput.mdx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+<AutoSizeInput prefix={<Icon name="search" />} />
+```
+
+<Preview>
+  <AutoSizeInput prefix={<Icon name="search" />} />
+</Preview>
+
+## Usage in forms with Field
+
+`Use `AutoSizeInput`with the`Field`component to get labels and descriptions. Also, you can use`AutoSizeInput`with the`required`attribute for validation. See the`Field` component for more information.
```
Comment (achatterjee-grafana) [inline packages/grafana-ui/src/components/AutoSizeInput/AutoSizeInput.mdx:24]:
```suggestion
Use `AutoSizeInput`with the`Field`component to get labels and descriptions. Also, you can use`AutoSizeInput`with the`required`attribute for validation. See the`Field` component for more information.
```

File: packages/grafana-ui/src/components/AutoSizeInput/AutoSizeInput.mdx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+```jsx
+<Field label="Important information" description="This information is very important, so you really need to fill it in">
+  <AutoSizeInput name="importantInput" required />
+</Field>
+```
+
+<Preview>
+  <Field
+    label="Important information"
+    description="This information is very important, so you really need to fill it in"
```
Comment (achatterjee-grafana) [inline packages/grafana-ui/src/components/AutoSizeInput/AutoSizeInput.mdx:35]:
```suggestion
    label="Important information!"
    description="Please enter the relevant information."
```

**Review by achatterjee-grafana — COMMENTED** [review]
Added a few comments.

**Review by ashharrison90 — APPROVED** [review]
lgtm 🙌

**Pull request merged**